src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -300,6 +305,52 @@ public void Encode<TPixel>(Image<TPixel> image, Stream stream)
             quantized?.Dispose();
         }
 
+        /// <summary>
+        /// Writes the transparency markers to the stream
+        /// </summary>
+        /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
+        /// <param name="pngMetaData">The image meta data.</param>
+        private void WriteTransparencyMarkers(Stream stream, PngMetaData pngMetaData)
+        {
+            if (pngMetaData.ColorType == PngColorType.Rgb)
+            {
+                if (pngMetaData.Rgb48Trans != null)
```

Comment:
We should be checking the bit depth here also.

Reply:
I'm not exactly sure how to do that, might be better if you implement this.

Reply:
Actually I think it's ok as we already throw if the combination is invalid.